### ISSUE TITLE
Colorbar overlay module with more advanced design

### ISSUE THREAD

**Issue by jywarren** [publiclab/image-sequencer#430]
After #285, we now can generate and overlay a colobar, and it looks great:

![](https://user-images.githubusercontent.com/25617855/43723773-13fce2e6-99b6-11e8-9926-f6e3133a76cb.png)

I'm thinking what the next step could be; we could potentially work on text overlay (currently in #165 but will be broken out) to show labels

and also display the colorbar with an outline (we also don't have "drawing rectangles" as a module... but we could! break this out too!)

The goal might be something like this:

![colorbar-image-sequencer](https://user-images.githubusercontent.com/24359/47272534-2ab15000-d555-11e8-923f-463a59e790cc.png)

We could also have the colorbar not be overlaid on the image, but shown below it. This could be done with the crop module maybe cropping at larger than the current canvas (although I'm not sure this will work -- would it leave white space?) https://github.com/publiclab/image-sequencer/blob/88364decf6a6111a0be31fd98010576d5c6f918e/src/modules/Crop/info.json

This could be broken into 3 different issues to solve.

**Comment by ho-dor** [comment]
can I work on this @jywarren  ?

**Comment by jywarren** [comment]
That would be super - we'd love your help!

**Comment by jywarren** [comment]
I've added a new issue for setting the background during a crop operation: https://github.com/publiclab/image-sequencer/issues/467

**Comment by jywarren** [comment]
So, this sequence is getting closer! https://sequencer.publiclab.org/examples/#steps=crop{x:0|y:0|w:611|h:479},gradient{},colormap{colormap:stretched},crop{x:0|y:0|w:611|h:60},resize{resize:25%25},overlay{x:5|y:5|offset:-4}

But I still can't get it to appear in the newly added padding below the image that results from the crop. But closer!

**Comment by Mridul97** [comment]
@tech4GT I would love to work on this issue!

**Comment by Mridul97** [comment]
I am thinking to implement drawing rectangles first, so that we can use it to display the colorbar with an outline!

**Comment by blurry-x-face** [comment]
@jywarren I made something like this

![newColorbar](https://user-images.githubusercontent.com/42492389/72879920-e1394d00-3d23-11ea-8ae2-4f7bee023c2d.png)

Is this what we require?
